src/marqo/tensor_search/validation.py

```diff
@@ -742,3 +742,25 @@ def validate_model_properties_no_model(model_properties: dict):
         
     return model_properties
 
+
+def validate_model_name_and_properties(index_settings):
+    """
+    Ensures that:
+    1. If search_model is given, then model must be given as well.
+    2. if model_properties is given, then model is given as well.
+    3. if search_model_properties is given, then search_model must be given as well.
+    """
+    model_name = index_settings[IndexSettingsField.index_defaults].get(IndexSettingsField.model)
+    model_properties = index_settings[IndexSettingsField.index_defaults].get(IndexSettingsField.model_properties)
+    search_model_name = index_settings[IndexSettingsField.index_defaults].get(IndexSettingsField.search_model)
+    search_model_properties = index_settings[IndexSettingsField.index_defaults].get(IndexSettingsField.search_model_properties)
+
+
+    if search_model_name is not None and model_name is None:
+        raise UnknownModelError(f"`search_model` cannot be specified without also specifying `model`. Received search_model={search_model_name} but found no `model`")
```

Comment:
All three look like they should be `InvalidArgError` as they are user input errors, not inference errors?

Reply:
You're right. I had handling in `tensor_search` to convert these `UnknownModelError` to `InvalidArgError`, but it's better to just have them here in the validation function itself.